python/lsst/ip/isr/overscan.py

```diff
@@ -0,0 +1,504 @@
+        correct integer median of row-values.
+
+        Parameters
+        ----------
+        maskedArray : `numpy.ma.masked_array`
+            Masked array of input overscan data.
+
+        Returns
+        -------
+        collapsed : `numpy.ma.masked_array`
+            Single dimensional overscan data, combined with the afwMath median.
+        """
+        integerMI = self.integerConvert(maskedArray)
+
+        collapsed = []
+        fitType = afwMath.stringToStatisticsProperty('MEDIAN')
+        for row in integerMI:
+            rowMedian = afwMath.makeStatistics(row, fitType, self.statControl).getValue()
+            collapsed.append(rowMedian)
+
+        return np.array(collapsed)
+
+    def splineFit(self, indices, collapsed, numBins):
+        """Wrapper function to match spline fit API to polynomial fit API.
+
+        Parameters
+        ----------
+        indices : `numpy.ndarray`
+            Locations to evaluate the spline.
+        collapsed : `numpy.ndarray`
+            Collapsed overscan values corresponding to the spline
+            evaluation points.
+        numBins : `int`
+            Number of bins to use in constructing the spline.
+
+        Returns
+        -------
+        interp : `lsst.afw.math.Interpolate`
+            Interpolation object for later evaluation.
+        """
+        if not np.ma.is_masked(collapsed):
+            collapsed.mask = np.array(len(collapsed)*[np.ma.nomask])
+
+        numPerBin, binEdges = np.histogram(indices, bins=numBins,
+                                           weights=1 - collapsed.mask.astype(int))
+        with np.errstate(invalid="ignore"):
+            values = np.histogram(indices, bins=numBins,
+                                  weights=collapsed.data*~collapsed.mask)[0]/numPerBin
+            binCenters = np.histogram(indices, bins=numBins,
+                                      weights=indices*~collapsed.mask)[0]/numPerBin
+            interp = afwMath.makeInterpolate(binCenters.astype(float)[numPerBin > 0],
+                                             values.astype(float)[numPerBin > 0],
+                                             afwMath.stringToInterpStyle(self.config.fitType))
+        return interp
+
+    @staticmethod
+    def splineEval(indices, interp):
+        """Wrapper function to match spline evaluation API to polynomial fit API.
+
+        Parameters
+        ----------
+        indices : `numpy.ndarray`
+            Locations to evaluate the spline.
+        interp : `lsst.afw.math.interpolate`
+            Interpolation object to use.
+
+        Returns
+        -------
+        values : `numpy.ndarray`
+            Evaluated spline values at each index.
+        """
+
+        return interp.interpolate(indices.astype(float))
+
+    @staticmethod
+    def maskExtrapolated(collapsed):
+        """Create mask if edges are extrapolated.
+
+        Parameters
+        ----------
+        collapsed : `numpy.ma.masked_array`
```

Comment:
Why the name `collapsed`? Isn't this just a 1D array?

Reply:
Yes, but it's the 1d array after the second dimension has been removed.